gtests/test_nopython_buffer.cpp

```diff
@@ -6,9 +6,22 @@
 #error "Python.h should not be included."
 #endif
 
-TEST(nopython_buffer, dummy)
+TEST(ConcreteBuffer, iterator)
 {
-    EXPECT_TRUE(true);
+    using namespace modmesh;
+
+    auto buffer = ConcreteBuffer::construct(10);
+    int8_t i = 0;
+    for (auto & it : *buffer)
+    {
+        it = i++;
+    }
+
+    i = 0;
+    for (const auto it : *buffer)
```

Comment:
testing const iterator

Reply:
Why the const iterator testing does not use reference while the non-const iterator testing does?

Reply:
I think both works, but my original thought is that the value here is `int8_t`, so copying it is more efficient than getting the reference from it?

Reply:
I see, the explanation is clear.  This is good.  I just wanted to check that it is intentional.